### PR TITLE
Fix invalid pointer types in ldloca/ldarga descriptions

### PR DESCRIPTION
**Pull request by MichalPetryka**
## Summary

Those are documented to push managed pointers on the stack in the ECMA 335 and [the runtime implements them as such](https://github.com/dotnet/runtime/blob/dd158ba513a5a09f8889e5cf8873655a4f3c6fdd/src/coreclr/jit/importer.cpp#L6639).

### CODE REVIEW

File: xml/System.Reflection.Emit/OpCodes.xml [hunk 1]
```diff
@@ -10065,7 +10065,7 @@ The following <xref:System.Reflection.Emit.ILGenerator.Emit%2A> method overloads
   
 1.  The address stored in the local variable at the specified index is pushed onto the stack.  
   
- The `ldloca` instruction pushes the address of the local variable number at the passed index onto the stack, where local variables are numbered 0 onwards. The value pushed on the stack is already aligned correctly for use with instructions like <xref:System.Reflection.Emit.OpCodes.Ldind_I> and <xref:System.Reflection.Emit.OpCodes.Stind_I>. The result is a transient pointer (type `*`).  
+ The `ldloca` instruction pushes the address of the local variable number at the passed index onto the stack, where local variables are numbered 0 onwards. The value pushed on the stack is already aligned correctly for use with instructions like <xref:System.Reflection.Emit.OpCodes.Ldind_I> and <xref:System.Reflection.Emit.OpCodes.Stind_I>. The result is a managed pointer (type `&`) which can however be safely converted to a transient pointer (type `*`) with <xref:System.Reflection.Emit.OpCodes.Conv_U>.
```
Comment (reflectronic) [inline xml/System.Reflection.Emit/OpCodes.xml:10068]:
```suggestion
 The `ldloca` instruction pushes the address of the local variable number at the passed index onto the stack, where local variables are numbered 0 onwards. The value pushed on the stack is already aligned correctly for use with instructions like <xref:System.Reflection.Emit.OpCodes.Ldind_I> and <xref:System.Reflection.Emit.OpCodes.Stind_I>. The result is a managed pointer (type `&amp;`). The local variable is stored in unmanaged memory, so the return value can be converted to an unmanaged pointer without pinning.
```

File: xml/System.Reflection.Emit/OpCodes.xml [hunk 2]
```diff
@@ -6583,7 +6583,7 @@
   
 1.  The address `addr` of the argument indexed by `index` is pushed onto the stack.  
   
- The `ldarga` instruction fetches the address (of type `*`) of the argument indexed by `index`, where arguments are indexed from 0 onwards. The address `addr` is always aligned to a natural boundary on the target machine.  
+ The `ldarga` instruction fetches the address (of type `&`, which can however be safely converted to type `*` with <xref:System.Reflection.Emit.OpCodes.Conv_U>) of the argument indexed by `index`, where arguments are indexed from 0 onwards. The address `addr` is always aligned to a natural boundary on the target machine.
```
Comment (reflectronic) [inline xml/System.Reflection.Emit/OpCodes.xml:6586]:
```suggestion
 The `ldarga` instruction fetches the address (of type `&amp;`) of the argument indexed by `index`, where arguments are indexed from 0 onwards. The address `addr` is always aligned to a natural boundary on the target machine. The argument is stored in unmanaged memory, so the return value can be converted to an unmanaged pointer without pinning.
```

**Review by reflectronic — COMMENTED** [review]
`conv.u` does not convert managed pointers to transient pointers (type `*`), so this patch is incorrect. The transient pointers concept was deleted from the standard recently in https://github.com/dotnet/runtime/pull/86622, so any reference to transient pointers should be deleted. There are many more references elsewhere in this file.

File: xml/System.Reflection.Emit/OpCodes.xml (hunk 1 above)
Comment (MichalPetryka) [reply xml/System.Reflection.Emit/OpCodes.xml:10068]: I think `conv.u` should still be mentioned there.

**Comment by gewarren** [comment]
Tagging @dotnet/area-system-reflection-emit

**Comment by steveharter** [comment]
PTAL @AaronRobinsonMSFT

File: xml/System.Reflection.Emit/OpCodes.xml (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply xml/System.Reflection.Emit/OpCodes.xml:10068]: I disagree. I would prefer to keep this as close to ECMA-335 modified with the relavantly applied addendums.

**Comment by gewarren** [comment]
@AaronRobinsonMSFT Is this how it should render?

image

**Comment by AaronRobinsonMSFT** [comment]
> @AaronRobinsonMSFT Is this how it should render?
> 
> image

Definitely not. Let me try again. :)

File: xml/System.Reflection.Emit/OpCodes.xml [hunk 3]
```diff
@@ -6583,7 +6583,7 @@
   
 1.  The address `addr` of the argument indexed by `index` is pushed onto the stack.  
   
- The `ldarga` instruction fetches the address (of type `*`) of the argument indexed by `index`, where arguments are indexed from 0 onwards. The address `addr` is always aligned to a natural boundary on the target machine.  
+ The `ldarga` instruction fetches the address (of type `&amp;`) of the argument indexed by `index`, where arguments are indexed from 0 onwards. The address `addr` is always aligned to a natural boundary on the target machine. The argument is stored in unmanaged memory, so the return value can be converted to an unmanaged pointer without pinning.
```
Comment (AaronRobinsonMSFT) [inline xml/System.Reflection.Emit/OpCodes.xml:6586]:
```suggestion
 The `ldarga` instruction fetches the address (of type `&`) of the argument indexed by `index`, where arguments are indexed from 0 onwards. The address `addr` is always aligned to a natural boundary on the target machine. The argument is stored in unmanaged memory, so the return value can be converted to an unmanaged pointer without pinning.
```

File: xml/System.Reflection.Emit/OpCodes.xml [hunk 4]
```diff
@@ -10065,7 +10065,7 @@ The following <xref:System.Reflection.Emit.ILGenerator.Emit%2A> method overloads
   
 1.  The address stored in the local variable at the specified index is pushed onto the stack.  
   
- The `ldloca` instruction pushes the address of the local variable number at the passed index onto the stack, where local variables are numbered 0 onwards. The value pushed on the stack is already aligned correctly for use with instructions like <xref:System.Reflection.Emit.OpCodes.Ldind_I> and <xref:System.Reflection.Emit.OpCodes.Stind_I>. The result is a transient pointer (type `*`).  
+ The `ldloca` instruction pushes the address of the local variable number at the passed index onto the stack, where local variables are numbered 0 onwards. The value pushed on the stack is already aligned correctly for use with instructions like <xref:System.Reflection.Emit.OpCodes.Ldind_I> and <xref:System.Reflection.Emit.OpCodes.Stind_I>. The result is a managed pointer (type `&amp;`). The local variable is stored in unmanaged memory, so the return value can be converted to an unmanaged pointer without pinning.
```
Comment (AaronRobinsonMSFT) [inline xml/System.Reflection.Emit/OpCodes.xml:10068]:
```suggestion
 The `ldloca` instruction pushes the address of the local variable number at the passed index onto the stack, where local variables are numbered 0 onwards. The value pushed on the stack is already aligned correctly for use with instructions like <xref:System.Reflection.Emit.OpCodes.Ldind_I> and <xref:System.Reflection.Emit.OpCodes.Stind_I>. The result is a managed pointer (type `&`). The local variable is stored in unmanaged memory, so the return value can be converted to an unmanaged pointer without pinning.
```

**Comment by AaronRobinsonMSFT** [comment]
@gewarren Should be fixed now.

**Review by gewarren — APPROVED** [review]
LGTM
